plugin/src/main/java/org/opensearch/ml/autoreload/MLModelAutoReDeployer.java

```diff
@@ -0,0 +1,269 @@
+        ActionListener<SearchResponse> listener = ActionListener.wrap(res -> {
+            if (res != null && res.getHits() != null && res.getHits().getTotalHits() != null && res.getHits().getTotalHits().value > 0) {
+                Arrays.stream(res.getHits().getHits()).filter(x -> x != null && x.getSourceAsMap() != null).forEach(x -> {
+                    ModelAutoRedeployArrangement modelAutoRedeployArrangement = ModelAutoRedeployArrangement
+                        .builder()
+                        .addedNodes(addedNodes)
+                        .searchResponse(x)
+                        .build();
+                    boolean notExist = modelAutoRedeployArrangements.stream().noneMatch(y -> y.equals(modelAutoRedeployArrangement));
+                    if (notExist)
+                        modelAutoRedeployArrangements.add(modelAutoRedeployArrangement);
+                });
+                redeployAModel();
+            }
+        }, e -> { log.error("Failed to query need auto redeploy models, no action will be performed, addedNodes are: {}", addedNodes, e); });
+
+        queryRunningModels(listener);
+    }
+
+    private void triggerUndeployModelsOnDataNodes(List<String> dataNodeIds) {
+        List<String> modelIds = new ArrayList<>();
+        ActionListener<SearchResponse> listener = ActionListener.wrap(res -> {
+            if (res != null && res.getHits() != null && res.getHits().getTotalHits() != null && res.getHits().getTotalHits().value > 0) {
+                Arrays.stream(res.getHits().getHits()).forEach(x -> modelIds.add(x.getId()));
+                if (modelIds.size() > 0) {
+                    ActionListener<MLUndeployModelNodesResponse> undeployModelListener = ActionListener
+                        .wrap(
+                            r -> { log.info("Undeploy models on data nodes successfully!"); },
+                            e -> { log.error("Failed to undeploy models on data nodes, error is: {}", e.getMessage(), e); }
+                        );
+                    MLUndeployModelNodesRequest undeployModelNodesRequest = new MLUndeployModelNodesRequest(
+                        dataNodeIds.toArray(new String[0]),
+                        modelIds.toArray(new String[0])
+                    );
+                    client.execute(MLUndeployModelAction.INSTANCE, undeployModelNodesRequest, undeployModelListener);
+                }
+            }
+        }, e -> { log.error("Failed to query need undeploy models, no action will be performed"); });
+        queryRunningModels(listener);
+    }
+
+    private void queryRunningModels(ActionListener<SearchResponse> listener) {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        QueryBuilder queryBuilder = QueryBuilders
+            .boolQuery()
+            .mustNot(QueryBuilders.existsQuery(MLModel.CHUNK_NUMBER_FIELD))
+            .should(QueryBuilders.termQuery(MLModel.MODEL_STATE_FIELD, MLModelState.DEPLOYING.name()))
```

Comment:
I think we use `TermsQueryBuilder` here, to filter multiple states

Reply:
changed to TermsQueryBuilder